services/app-api/src/postgres/contractAndRates/prismaSharedContractRateHelpers.ts

```diff
@@ -60,6 +63,16 @@ const includeRateFormData = {
     supportingDocuments: true,
     certifyingActuaryContacts: true,
     addtlActuaryContacts: true,
+    contractsWithSharedRateRevision: {
+        include: {
+            revisions: {
+                take: 1,
```

Comment:
This looks fine to me based on the prisma table -  to answer your question from the PR description. I think returning just latest rev for the linked contract makes sense.

 I think it _could_ be possible then we could display too early  the updated programs from a rate that is unlocked and being edited by the state ...but I don't see a way around that since its a contract - rate revision relationship and not a contract revision - rate revision level relationship right now.